src/main/java/org/jetbrains/research/intellijdeodorant/core/ast/MethodObject.java

```diff
@@ -248,18 +256,23 @@ public MethodInvocationObject isDelegate() {
                                 }
                             }
                         }
-                    } else if (!PsiTreeUtil.findChildrenOfType(methodInvocationExpression, PsiReferenceExpression.class).isEmpty()) {
-                        Collection<PsiReferenceExpression> references = PsiTreeUtil.findChildrenOfType(methodInvocationExpression, PsiReferenceExpression.class);
-                        for (PsiReferenceExpression reference : references) {
-                            PsiElement resolvedReference = reference.resolve();
-                            if (resolvedReference instanceof PsiField) {
-                                PsiField psiField = (PsiField) reference.resolve();
-                                if (psiField != null && psiField.getContainingClass() != null && psiField.getContainingClass().equals(parentClass)
-                                        || psiField != null && parentClass != null && psiField.getContainingClass() != null && parentClass.isInheritor(psiField.getContainingClass(), true)) {
-                                    for (MethodInvocationObject methodInvocationObject : methodInvocations) {
-                                        if (methodInvocationExpression.equals(methodInvocation.getMethodExpression())) {
-                                            return methodInvocationObject;
-                                        }
+                    } else if (resolved instanceof PsiField) {
+                        PsiField variableBinding = (PsiField) resolved;
+
+                        if (variableBinding.getContainingClass().equals(parentClass) ||
+                                parentClass.isInheritor(variableBinding.getContainingClass(), true)) {
+                            for (MethodInvocationObject methodInvocationObject : methodInvocations) {
+                                if (methodInvocationObject.getMethodInvocation().equals(methodInvocation)) {
+                                    return methodInvocationObject;
+                                }
+                            }
+                        }
+                    } else if (resolved != null) {
```

Comment:
I think you could simplify it to` if (resolved instanceof PsiParameter)`

Reply:
I can remove `if (resolved instanceof PsiVariable)` (done), but not `if (resolvedElement != null)` (or we could go into other branches if it is not null, but not instance of PsiParameter either).